### DISCUSSION TITLE
how to sent onchain massage one address to another address

### DISCUSSION THREAD

**Discussion by dev-Crypto360** [Cyfrin/foundry-full-course-cu#926] [Q&A]
hi guys,
I am really facing an issue I really don't understand how to sent a massage from 1 address to another address like think I want to sent a SMS to a hacker who stole my money from my metamask to his metamask I have only his eth address I even don't know him ,
so in this situation how to I sent a sms to this hacker address to give my money back .

if anyone understand me kindly give me answer if possible give me the code also which I needed . 

Note : this hacker think just a example .

Thankyou in advanced ❤️❤️❤️

**Comment by theirrationalone** [comment]
IMO, chatting with others into blockchain could cost us Gas ⛽. 

Another disadvantage could be unseen msg. If we try to deploy a SMCT in that pointing to hacker's address/EOA/Wallet, with some msg. It would be like a niddle into the wild. Since it's a decentralized world who knows who have seen our contract and its txs events. 

One heck could be.... Save hacker's address and try to find him/her into WEB3 hacking contests Or somewhere else where ever you could. If your luck worked then you will get at least his/her social info. After that try to chat with him/her. If somehow he/she claimed anywhere that the address/EOA/WALLET which has stolen your money belongs to him/her then you will be able to raise a dispute against him/her with all legit proofs. 

But i don't think a hacker could be as silly as i described above. 

If you lost your money, Even it would almost be an impossible to interact with him/her. However deploying a SMCT to chat with him/her, would a waste of money. 

Bitter but It's the bare truth. 😔

Sorry if i'm wrong here. 

;`)

**Reply by dev-Crypto360** [reply]
I just want to know how this massage function work as a learner I think I need to learn all this Things, that's why I was curious about it after hearing HTX exchange hack ! 

So I can sent sms using SMCT tools? 

Don't Warry I will try tx on Testnet 😂
